azkaban-web-server/src/web/js/azkaban/view/flow-execution-list.js

```diff
@@ -266,26 +264,53 @@ azkaban.ExecutionListView = Backbone.View.extend({
     }
   },
 
-  toggleExpandFlow: function (flow) {
-    console.log("Toggle Expand");
+  /**
+   * Expands or collapses a flow node according to the value of the expand
+   * parameter.
+   *
+   * @param expand - if value true -> expand, false: collapse,
+   *                 undefined -> toggles node status
+   */
+  setFlowExpansion: function (flow, expand) {
     var tr = flow.joblistrow;
     var subFlowRow = tr.subflowrow;
     var expandIcon = $(tr).find("> td > .listExpand");
-    if (tr.expanded) {
-      tr.expanded = false;
-      $(expandIcon).removeClass("glyphicon-chevron-up");
-      $(expandIcon).addClass("glyphicon-chevron-down");
 
-      $(tr).removeClass("expanded");
-      $(subFlowRow).hide();
-    }
-    else {
+    var needs_expansion = !tr.expanded && (expand === undefined || expand);
+    var needs_collapsing = tr.expanded && (expand === undefined || !expand);
+
+    if (needs_expansion) {
       tr.expanded = true;
       $(expandIcon).addClass("glyphicon-chevron-up");
       $(expandIcon).removeClass("glyphicon-chevron-down");
       $(tr).addClass("expanded");
       $(subFlowRow).show();
+
+    } else if (needs_collapsing) {
+      tr.expanded = false;
+      $(expandIcon).removeClass("glyphicon-chevron-up");
+      $(expandIcon).addClass("glyphicon-chevron-down");
+      $(tr).removeClass("expanded");
+      $(subFlowRow).hide();
+    } // else do nothing
+  },
+
+  expandFailedJobs: function (nodes) {
+    var failed = false;
+    for (var i = 0; i < nodes.length; ++i) {
+      var node = nodes[i].changedNode ? nodes[i].changedNode : nodes[i];
```

Comment:
With this we are passing an empty array whenever there is no `nodes` property in `node` or it has null value. We are doing this to prevent "Cannot read property 'length' of undefined/null" error in the for loop. It’s mainly "just is case" and it can be removed if we wanted because Azkaban doesn’t allow to have an empty flow.